src/react/deckgl.js

```diff
@@ -41,6 +41,12 @@ export default class DeckGL extends React.Component {
     }
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    const transitionTriggered = this.deck.triggerViewportTransition(nextProps);
```

Comment:
There are many props that can change here that are not related to transitions. 

Coded this way there seems to be a risk that we decline some non-transition related updates that we should be doing.

Think about how we can make the logic clearer here? Or add more comments explaining why this is not an issue?